pkg/adapter/info.go

```diff
@@ -23,6 +23,9 @@ import (
 type BuilderInfo struct {
 	// Name returns the official name of the adapter.
 	Name string
+	// ShortName is used to register this type with the config system.
```

Comment:
Let's not add this field. We should just systematically derive the short name, no need to put the burden on the developer. Even if it is optional, the field would show up in docs. Since we can derive it, let's just do that all the time. We can always introduce this field later is there's a reason to allow the developer to control this name.

And in addition, I don't think calling this "ShortName" is correct. The name is actually just as long. ResourceTypeName or something may be more appropriate.